eitprocessing/eit_data/__init__.py

```diff
@@ -35,27 +32,24 @@ class EITData(SelectByTime, Equivalence, ABC):
     phases: list = field(default_factory=list)
     events: list = field(default_factory=list)
     label: str | None = None
-    variants: VariantCollection = field(
-        default_factory=lambda: VariantCollection(EITDataVariant)
-    )
+    pixel_impedance: NDArray = field(repr=False, kw_only=True)
 
     def __post_init__(self):
         if not self.label:
             self.label = f"{self.__class__.__name__}_{id(self)}"
         self._check_equivalence = ["vendor", "framerate"]
 
     @classmethod
-    def from_path(  # pylint: disable=too-many-arguments,too-many-locals
+    def from_path(
         cls,
         path: PathArg,
         vendor: Vendor | str,
-        label: str | None = None,
         framerate: float | None = None,
         first_frame: int = 0,
         max_frames: int | None = None,
         return_non_eit_data: bool = False,
-    ) -> Self | tuple[Self, ContinuousDataCollection, SparseDataCollection]:
-        """Load sequence from path(s)
+    ) -> DataCollection | tuple[DataCollection, DataCollection, DataCollection]:
+        """Load sequence from path(s).
```

Comment:
Agreed. 

`from_path()` returns either a DataCollection(EITData), or a tuple containing DataCollection(EITData), DataCollection(ContinuousData), DataCollection(SparseData). If there is no sparse data, the third DataCollection has no items. 

This whole way of loading data is bothering me. It adds a lot of complexity (as you rightly point out). I think a good solution would be to separate loading data from the classes, as is common practice. But this would require yet another rewrite and changes to the GUI. It is something to consider/discuss next week?

Reply:
How do you feel about naming this argument `attach_additional_data` or `attach_external_data` or something like that?